cloud/ec2_fleet.go

```diff
@@ -0,0 +1,538 @@
+package cloud
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+type ec2FleetManager struct {
+	client      AWSClient
+	credentials *credentials.Credentials
+	settings    *evergreen.Settings
+}
+
+func (m *ec2FleetManager) GetSettings() ProviderSettings {
+	return &EC2ProviderSettings{}
+}
+
+func (m *ec2FleetManager) Configure(ctx context.Context, settings *evergreen.Settings) error {
+	if m.client == nil {
```

Comment:
Will it ever not be nil? Obviously it's fine to check this, but it might be confusing for a later reader, since it suggests that we're caching the `awsClientImpl`, but in fact I assume that `m.client` is nil every time we call `Configure`. I haven't read code to confirm this though.

Reply:
removed.